### ISSUE TITLE
SC2086 ("Double quote to prevent globbing and word splitting.") should not apply to variables declared to be integers

### ISSUE THREAD

**Issue by srcshelton** [koalaman/shellcheck#289]
If a variable is declared to be numeric (with declare/local -i) then SC2086 should not apply - the code:

```sh
function foo() {
    local -i rc=${?}

    ...

    exit ${rc}
}
```

... should not generate a warning.

**Comment by Speeddymon** [comment]
I agree with the sentiment, however the example does not go far enough to demonstrate the issue.

```
declare -i RC
function handle_rc()
{
        local -i rc
        rc=$1

        if [ $rc -ne 0 ]
        then
                RC=$rc
        fi
        return $RC
}
handle_rc 1
```

**Comment by a1346054** [comment]
See #2341 for why this isn't a false positive.

This issue can be closed.

**Comment by kurahaupo** [comment]
This seems like the sort of stylistic choice that cannot be made to satisfy everyone.

(I've seen code in the wild that does `IFS=89` in an attempt to avoid complaints about broken octal numbers.)

Maybe we need a option `--assume-sane-ifs` that assumes that IFS won't contain `[-+0-9_a-zA-Z]`, and therefore it is safe to leave unquoted the `$((...))` expansion, the built-in numeric variables, and any variables that are marked with `-i`.
